app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -859,11 +860,11 @@ class GleanMetricsService(
 
             // setStartupMetrics is not a fast function. It does not need to be done before we can consider
             // ourselves initialized. So, let's do it, well, later.
-            setStartupMetrics()
+            setStartupMetrics(context.settings())
         }
     }
 
-    internal fun setStartupMetrics() {
+    internal fun setStartupMetrics(settings: Settings) {
```

Comment:
I am curious if it would've made any difference if we just set `val settings = context.settings()` instead of passing `settings` or is it must always `settings` as a parameter and if there was any particular reason that you found from testing.

Reply:
I needed to pass this in for testing.  Otherwise there's not good way to use a mock.